tsuba: remove tsuba::Stat

tylershunt opened the pull request:
Removed because `tsuba::ListAvailableViews` accomplishes the same thing.
I cleaned up and tweaked ListAvailableViews as well.

The semantics have been tweaked so that a view is returned for the
version 0 of the graph. This "view" (if you can even call it that) is
special in that it's not a valid RDG that can be loaded, but returning
it is a nice way to signal to callers that the graph is empty.

JIRA: KAT-1284

Signed-off-by: Tyler Hunt <thunt@katanagraph.com>

gurbinder533 commented on scripts/check_bipart.sh:
```diff
@@ -0,0 +1,29 @@
+    for input in "${INPUTS[@]}"; do
+      for partition in "${PARTITIONS[@]}"; do
+        statfname=${input}_p${partition}_thrd${thread}_run${run}
+        cmd="./bipart-cpu "${INPUT_DIR}"/"$input" 25 2 "
```
Use an absolute path. Have a variable point to the location of executable. ${EXE_PATH}/bipart-cpu

gurbinder533 replied:
Also, the arguments have changed. The number of partitions are no longer positional arguments. Please check all the arguments.

gurbinder533 commented on scripts/check_bipart.sh:
```diff
@@ -0,0 +1,29 @@
+#Note that this is also a list.
+PARTITIONS=("16")
+#Input directory having the above specified input graphs.
+INPUT_DIR="/home/hochan/inputs"
```
Make this is a command-line argument: INPUT_DIR=$1 and EXE_PATH=$2

gurbinder533 commented:
@uditagar Can you please merge this?

uditagar commented:
> @uditagar Can you please merge this?

I will fix this this weekend. Thanks

Commit:
tsuba: remove tsuba::Stat

Removed because `tsuba::ListAvailableViews` accomplishes the same thing.
I cleaned up and tweaked ListAvailableViews as well.

The semantics have been tweaked so that a view is returned for the
version 0 of the graph. This "view" (if you can even call it that) is
special in that it's not a valid RDG that can be loaded, but returning
it is a nice way to signal to callers that the graph is empty.

JIRA: KAT-1284

Signed-off-by: Tyler Hunt <thunt@katanagraph.com>

tylershunt commented:
enterprise fixed in https://github.com/KatanaGraph/katana-enterprise/pull/1848

yshalabi commented:
There’s been a few conversations about what version zero signifies and how to represent an empty graph. My preference is to continue the invalidity of version zero RDGs and to add routines to create a new empty RDG which starts at version 1.

tylershunt commented:
> There’s been a few conversations about what version zero signifies and how to represent an empty graph. My preference is to continue the invalidity of version zero RDGs and to add routines to create a new empty RDG which starts at version 1.

This is still true; you can't load a valid RDG from version 0. I could return an empty list instead; that's fine with me; this is just a convenient way to distinguish:
 * storage location created but nothing has been stored there yet
 * storage location does not exist

ddn0 commented:
> > There’s been a few conversations about what version zero signifies and how to represent an empty graph. My preference is to continue the invalidity of version zero RDGs and to add routines to create a new empty RDG which starts at version 1.
> 
> This is still true; you can't load a valid RDG from version 0. I could return an empty list instead; that's fine with me; this is just a convenient way to distinguish:
> 
>     * storage location created but nothing has been stored there yet
> 
>     * storage location does not exist

I think we are close to getting the sensible semantics. The source of our difficulties is that RDGs have a storage location and are loaded, so we haven't clearly delineated the RDG that we will write (for the initial version).

I see now that we can have zero views for a version. Is that only for the newly created RDGs or could that happen for other RDG versions? I can see a possibility of having no views (potentially) in the process of creating the first view of a version or after some extreme vacuuming of old views.

tylershunt commented:
> I see now that we can have zero views for a version. Is that only for the newly created RDGs or could that happen for other RDG versions? I can see a possibility of having no views (potentially) in the process of creating the first view of a version or after some extreme vacuuming of old views.

I think it makes sense to allow zero views for any version as a special-case representation for an empty graph. My reasoning is that it seems trivial to construct an empty graph in whatever view you want; so it seems convenient to have some sentinel for empty that is semantically "any view".
